llvm/lib/Target/AMDGPU/AMDGPUInsertDelayAlu.cpp

```diff
@@ -331,6 +331,11 @@ class AMDGPUInsertDelayAlu {
     bool Changed = false;
     MachineInstr *LastDelayAlu = nullptr;
 
+    bool VALUSALUStall = false;
+    MCRegUnit lastSgprWrite = 0;
+    MCRegUnit longestWait = 0;
+    unsigned deletedCyclesNum = 0;
```

Comment:
Reduce the scope of variables as much as possible.

For example, VALUSALUStall doesn't need to be defined outside the loop.